monix-execution/jvm/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -0,0 +1,104 @@
+package monix.execution.misc
+
+import java.util.UUID
+
+object Local {
+
+  type Context = scala.collection.immutable.Map[String, Option[LocalContext]]
+
+  trait LocalContext
+
+  private[this] def context(k: String, v: Option[LocalContext]): Context =
+    scala.collection.immutable.Map(k -> v)
+
+  private[this] val localContext = ThreadLocal[Context]()
+
+  private def register(): String = synchronized {
+    UUID.randomUUID().toString
```

Comment:
This seems a little wasteful, with the synchronization and the `toString`. Do we really need to use strings for IDs btw? Strings are awful.

Reply:
@alexandru I will change it. I will follow the same implementation as finagle where they use class instances for that. Do you agree?